xdsl/interpreters/onnx.py

```diff
@@ -0,0 +1,93 @@
+import subprocess
+import sys
+from typing import Any, cast
+
+from xdsl.dialects import onnx
+from xdsl.interpreter import (
+    Interpreter,
+    InterpreterFunctions,
+    impl,
+    register_impls,
+)
+from xdsl.interpreters.shaped_array import ShapedArray
+
+try:
+    import numpy as np
+except (ImportError, ModuleNotFoundError):
+    np = None
+
+if np is None:
+    # if numpy is not available, uninstall onnx
+    try:
+        subprocess.check_call([sys.executable, "-m", "pip", "uninstall", "-y", "onnx"])
+        print("ONNX uninstallation successful.")
+    except subprocess.CalledProcessError:
+        print("ONNX uninstallation unsuccessful. Please uninstall it manually.")
+
+# class happens only if np is available
+if np is not None:
+
+    @register_impls
+    class OnnxFunctions(InterpreterFunctions):
+        @impl(onnx.Add)
+        def run_add(
+            self, interpreter: Interpreter, op: onnx.Add, args: tuple[Any, ...]
+        ) -> tuple[Any, ...]:
+            lhs, rhs = args[0], args[1]
+            assert isinstance(lhs, ShapedArray)
+            assert isinstance(rhs, ShapedArray)
+            lhs = cast(ShapedArray[int], lhs)
+            rhs = cast(ShapedArray[int], rhs)
+            assert lhs.shape == rhs.shape
+            result = [l + r for l, r in zip(np.array(lhs.data), np.array(rhs.data))]
+            return (ShapedArray(result, lhs.shape),)
+
+        @impl(onnx.Sub)
+        def run_sub(
+            self, interpreter: Interpreter, op: onnx.Sub, args: tuple[Any, ...]
+        ) -> tuple[Any, ...]:
+            lhs, rhs = args[0], args[1]
+            assert isinstance(lhs, ShapedArray)
+            assert isinstance(rhs, ShapedArray)
+            lhs = cast(ShapedArray[int], lhs)
+            rhs = cast(ShapedArray[int], rhs)
+            assert lhs.shape == rhs.shape
+            result = [l - r for l, r in zip(np.array(lhs.data), np.array(rhs.data))]
```

Comment:
```suggestion
            result = np.array(lhs.data) - np.array(rhs.data)
```

Reply:
this messes up the tests for some reason